bgpd/bgp_route.c

```diff
@@ -4609,10 +4609,11 @@ void bgp_update(struct peer *peer, const struct prefix *p, uint32_t addpath_id,
 
 		/* Nexthop reachability check - for unicast and
 		 * labeled-unicast.. */
-		if (((afi == AFI_IP || afi == AFI_IP6)
-		    && (safi == SAFI_UNICAST || safi == SAFI_LABELED_UNICAST))
-		    || (safi == SAFI_EVPN &&
-			bgp_evpn_is_prefix_nht_supported(p))) {
+		if (((afi == AFI_IP || afi == AFI_IP6) &&
+		     (safi == SAFI_UNICAST || safi == SAFI_LABELED_UNICAST)) ||
+		    (safi == SAFI_EVPN &&
+		     bgp_evpn_is_prefix_nht_supported(p)) ||
+		    afi == AFI_LINKSTATE) {
```

Comment:
there is a second place where nexthop reachability check should happen.
to illustrate, see https://github.com/FRRouting/frr/pull/13380/commits/fa8a9f2bb54253aabb2f5e7a59da851081e13f9c#diff-916a077bdc3f40b019663d23796308dd9ba7648f0a082bb975428b305c54b699R4782

Reply:
well seen !